ocean-components/src/main/java/br/com/useblu/oceands/components/compose/OceanCarouselWithComponents.kt

```diff
@@ -0,0 +1,165 @@
+import br.com.useblu.oceands.model.compose.carouselwithcomponents.OceanCarouselComponentItem
+import br.com.useblu.oceands.model.compose.carouselwithcomponents.OceanCarouselCycle
+import br.com.useblu.oceands.model.compose.carouselwithcomponents.OceanCarouselIndicator
+import br.com.useblu.oceands.ui.compose.OceanColors
+import br.com.useblu.oceands.ui.compose.OceanSpacing
+import kotlinx.coroutines.delay
+
+@Composable
+fun OceanCarouselWithComponents(
+    modifier: Modifier = Modifier,
+    pagerModifier: Modifier = Modifier,
+    items: Array<OceanCarouselComponentItem>,
+    indicator: OceanCarouselIndicator = OceanCarouselIndicator.PAGE,
+    cycle: OceanCarouselCycle = OceanCarouselCycle.Manual,
+    initialPage: Int = 0
+) {
+    val pagerState = rememberPagerState(
+        initialPage = initialPage,
+        pageCount = { items.size }
+    )
+
+    CarouselCycle(
+        cycle = cycle,
+        pagerState = pagerState
+    )
+
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .background(color = OceanColors.interfaceLightPure)
+    ) {
+        HorizontalPager(
+            state = pagerState,
+            contentPadding = PaddingValues(horizontal = OceanSpacing.xs),
+            pageSpacing = OceanSpacing.xxs,
+            modifier = pagerModifier
+        ) { page ->
+            if (LocalInspectionMode.current) {
+                Box(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .clickable { items[page].action() }
+                ) {
+                    Image(
+                        painter = painterResource(id = R.drawable.image_placeholder),
+                        contentDescription = null,
+                        modifier = Modifier.fillMaxSize(),
+                        contentScale = ContentScale.FillBounds
+                    )
+                }
+            } else {
+                Box(
+                    modifier = Modifier.clickable { items[page].action() },
+                    contentAlignment = Alignment.Center
+                ) {
+                    items[page].component()
+                }
+            }
+        }
+
+        PageIndicator(
+            type = indicator,
+            pages = items.size,
+            pagerState = pagerState
+        )
+    }
+}
+
+@Composable
+private fun CarouselCycle(
+    cycle: OceanCarouselCycle,
+    pagerState: PagerState,
+) {
+    LaunchedEffect(key1 = cycle) {
+        when (cycle) {
+            is OceanCarouselCycle.Manual -> {
+                return@LaunchedEffect
+            }
+            is OceanCarouselCycle.Auto -> {
+                while (true) {
+                    runCatching {
+                        delay(timeMillis = cycle.time)
+                        pagerState.animateScrollToPage(
+                            page = (pagerState.currentPage + 1) % pagerState.pageCount
+                        )
+                    }.getOrThrow()
```

Comment:
aqui tu nao ta usando esse valor pra nada, entao acho que nem precisa do getOrThrow
inclusive, como tu usou o getOrThrow, se desse uma exception, ele ainda iria propagar pra fora, se precisar do valor tu pode usar o getOrNull

```suggestion
                    }
```

Reply:
Peguei ali a mesma ideia do carousel com imagens, porém tá gerando, aqui, ANR quando sai da Activity. Acredito que esse padrão de repetição acaba que pode ser executado ainda depois do fim do Activity e aí gere o ANR. Dessa forma, mesmo que o throw ocorra ainda sem uma forma de lidar, não impede a continuidade.

Reply:
Vou tentar uma outra forma de fazer o update da page

Reply:
Hmmm, saquei

Reply:
Atualizei o fluxo removendo o ```while (true) { ``` e dessa forma não vem ocorrendo mais o ANR